docs/articles/documentation/test-api/obtaining-data-from-the-client/examples-of-using-client-functions.md

```diff
@@ -0,0 +1,174 @@
+
+To obtain a browser alias, create a client function that uses the [navigator.userAgent](https://www.w3schools.com/jsref/prop_nav_useragent.asp) property. The function will return a user-agent header sent by the browser. To parse the returned header, you can use external modules, for example, [ua-parser-js](https://github.com/faisalman/ua-parser-js).
+
+The following sample demonstrates how to create conditional test logic based on the used browser. Before running this test, install *ua-parser-js* by running `npm install ua-parser-js`.
+
+```js
+import { ClientFunction } from 'testcafe';
+import uaParser from 'ua-parser-js';
+
+fixture `My fixture`
+   .page `http://example.com`;
+
+//Returns a user-agent header sent by the browser
+const getUA = ClientFunction(() => navigator.userAgent);
+
+test('My Test', async t => {
+    const ua = await getUA();
+    const browserAlias =uaParser(ua).browser.name;
+
+    //Some common actions
+
+    //Conditional logic
+    if (browserAlias === 'Chrome'){
+        //Test logic for Chrome
+        console.log('The browser is Chrome');
+    } else if (browserAlias === 'Safari'){
+        //Test logic for Safari
+        console.log('The browser is Safari');
+    }
+});
+```
+
+## Injecting External Libraries into a Page from a Test
+
+One more case of using client functions is injecting external libraries like [jQuery](https://jquery.com/) into a tested page.
+
+In order to work with jQuery from a test, you first need to add jQuery code to the tested page by using the `eval` method.
+
+```js
+await t.eval(new Function(fs.readFileSync('./jquery.js').toString()));
+```
+
+After that,  you can use jQuery in the test within a client function.
+
+```js
+import { ClientFunction } from 'testcafe';
+import fs from 'fs';
+
+fixture `My fixture`
+   .page `http://devexpress.github.io/testcafe/example/`;
+
+test('My test', async t => {
+    // Adds jQuery code to the page
+    await t.eval(new Function(fs.readFileSync('./jquery.js').toString()));
+
+    const clientFunction = ClientFunction(() => {
+        // Uses the jQuery selector
+        return $('div.column.col-2').text();
+    });
+
+    const text = await clientFunction();
+    console.log(text);
+});
+```
+
+## Accessing Child Nodes in the DOM Hierarchy
+
+To find all child elements (not [nodes](https://developer.mozilla.org/en-US/docs/Web/API/Node)) of all nodes within a DOM hierarchy, TestCafe provides the [Selector.child](../selecting-page-elements/selectors/functional-style-selectors.md#child) method. However, you may need to access exactly child nodes of some node. You can do this by using a client function.
```

Comment:
1. I suspect that `all child elements of all nodes within a DOM hierarchy` include literally all elements on the page. That's not what we are going to `find`. Maybe you wanted to say that we can `search through` all these elements? And, BTW, I would say `all nodes in the DOM hierarchy` - not `within a`
2. I can't figure out what `exactly child nodes` means. Child nodes of an exact parent node?